flink-table/flink-sql-client/bin/sql-client.sh

```diff
@@ -58,18 +58,25 @@ log_setting=(-Dlog.file="$log" -Dlog4j.configuration=file:"$FLINK_CONF_DIR"/log4
 
 # get path of jar in /opt if it exist
 FLINK_SQL_CLIENT_JAR=$(find "$FLINK_OPT_DIR" -regex ".*flink-sql-client.*.jar")
+FLINK_PYTHON_JAR=$(find "$FLINK_OPT_DIR" -regex ".*flink-python.*.jar")
 
 # check if SQL client is already in classpath and must not be shipped manually
-if [[ "$CC_CLASSPATH" =~ .*flink-sql-client.*.jar ]]; then
+if [[ "$CC_CLASSPATH" =~ .*flink-sql-client.*.jar && "$CC_CLASSPATH" =~ .*flink-python.*.jar ]]; then
 
     # start client without jar
     exec $JAVA_RUN $JVM_ARGS "${log_setting[@]}" -classpath "`manglePathList "$CC_CLASSPATH:$INTERNAL_HADOOP_CLASSPATHS"`" org.apache.flink.table.client.SqlClient "$@"
 
 # check if SQL client jar is in /opt
 elif [ -n "$FLINK_SQL_CLIENT_JAR" ]; then
```

Comment:
In this case, maybe it's because `flink-sql-client` is in the classpath and `flink-python` is not.